// stencil-project-bench/dist/collection/components/j-button/j-button.js
import { h } from '@stencil/core';
import classNames from 'classnames';
export class JButton {
  constructor() {
    this.handleClick = () => {
      this.jClick.emit(true);
    };
    this.disabled = false;
    this.variant = 'contained';
    this.type = undefined;
    this.isDraggable = false;
  }
  setClassNames() {
    const { variant } = this;
    return classNames('j-button', variant);
  }
  render() {
    const { type, disabled, isDraggable } = this;
    return (h("button", { type: type, disabled: disabled, onClick: this.handleClick, class: this.setClassNames(), draggable: isDraggable }, h("slot", null)));
  }
  static get is() { return "j-button"; }
  static get encapsulation() { return "shadow"; }
  static get originalStyleUrls() {
    return {
      "$": ["j-button.scss"]
    };
  }
  static get styleUrls() {
    return {
      "$": ["j-button.css"]
    };
  }
  static get properties() {
    return {
      "disabled": {
        "type": "boolean",
        "mutable": false,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "disabled",
        "reflect": false,
        "defaultValue": "false"
      },
      "variant": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "'contained' | 'outlined'",
          "resolved": "\"contained\" | \"outlined\"",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "variant",
        "reflect": false,
        "defaultValue": "'contained'"
      },
      "type": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "type",
        "reflect": false
      },
      "isDraggable": {
        "type": "boolean",
        "mutable": false,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "is-draggable",
        "reflect": false,
        "defaultValue": "false"
      }
    };
  }
  static get events() {
    return [{
        "method": "jClick",
        "name": "jClick",
        "bubbles": true,
        "cancelable": true,
        "composed": true,
        "docs": {
          "tags": [],
          "text": ""
        },
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        }
      }];
  }
}
