// stencil-project-bench/dist/cjs/j-button.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-e5696325.js');

function createCommonjsModule(fn, basedir, module) {
	return module = {
		path: basedir,
		exports: {},
		require: function (path, base) {
			return commonjsRequire();
		}
	}, fn(module, module.exports), module.exports;
}

function commonjsRequire () {
	throw new Error('Dynamic requires are not currently supported by @rollup/plugin-commonjs');
}

var classnames = createCommonjsModule(function (module) {
/*!
	Copyright (c) 2018 Jed Watson.
	Licensed under the MIT License (MIT), see
	http://jedwatson.github.io/classnames
*/
/* global define */

(function () {

	var hasOwn = {}.hasOwnProperty;

	function classNames() {
		var classes = [];

		for (var i = 0; i < arguments.length; i++) {
			var arg = arguments[i];
			if (!arg) continue;

			var argType = typeof arg;

			if (argType === 'string' || argType === 'number') {
				classes.push(arg);
			} else if (Array.isArray(arg)) {
				if (arg.length) {
					var inner = classNames.apply(null, arg);
					if (inner) {
						classes.push(inner);
					}
				}
			} else if (argType === 'object') {
				if (arg.toString !== Object.prototype.toString && !arg.toString.toString().includes('[native code]')) {
					classes.push(arg.toString());
					continue;
				}

				for (var key in arg) {
					if (hasOwn.call(arg, key) && arg[key]) {
						classes.push(key);
					}
				}
			}
		}

		return classes.join(' ');
	}

	if (module.exports) {
		classNames.default = classNames;
		module.exports = classNames;
	} else {
		window.classNames = classNames;
	}
}());
});

const jButtonCss = ".j-button{cursor:pointer;background-color:#e9e9ee;border:1px solid #dbdbdb;padding:0.5rem 1rem;border-radius:0.5rem}.j-button.contained{background-color:#dfdfdf;border:1px solid #dbdbdb}.j-button.outlined{background-color:transparent;border:1px solid #c3c3c3}.j-button:hover{background-color:#e4e4e4;border:1px solid #dbdbdb}.j-button:active{background-color:#cacaca;border:1px solid #dbdbdb}";

const JButton = class {
  constructor(hostRef) {
    index.registerInstance(this, hostRef);
    this.jClick = index.createEvent(this, "jClick", 7);
    this.handleClick = () => {
      this.jClick.emit(true);
    };
    this.disabled = false;
    this.variant = 'contained';
    this.type = undefined;
    this.isDraggable = false;
  }
  setClassNames() {
    const { variant } = this;
    return classnames('j-button', variant);
  }
  render() {
    const { type, disabled, isDraggable } = this;
    return (index.h("button", { type: type, disabled: disabled, onClick: this.handleClick, class: this.setClassNames(), draggable: isDraggable }, index.h("slot", null)));
  }
};
JButton.style = jButtonCss;

exports.j_button = JButton;
